packages/interchain_queries/src/types.rs

```diff
@@ -140,17 +140,31 @@ impl KVReconstruct for Delegations {
         let mut delegations: Vec<cosmwasm_std::Delegation> = vec![];
 
         // first StorageValue is denom
+        if storage_values[0].value.is_empty() {
+            return Err(ContractError::InvalidQueryResultFormat(
+                "denom is empty".into(),
+            ));
+        }
         let denom: String = from_binary(&storage_values[0].value)?;
 
         // the rest are delegations and validators alternately
         for chunk in storage_values[1..].chunks(2) {
+            if chunk[0].value.is_empty() {
+                continue;
+            }
             let delegation_sdk: Delegation = Delegation::decode(chunk[0].value.as_slice())?;
+
             let mut delegation_std = cosmwasm_std::Delegation {
                 delegator: Addr::unchecked(delegation_sdk.delegator_address.as_str()),
                 validator: delegation_sdk.validator_address,
                 amount: Default::default(),
             };
 
+            if chunk[1].value.is_empty() {
```

Comment:
talking about corrupted data there may be an even length of `storage_values`  -> odd length of `storage_values[1..]` -> last chunk does not have [1] element

Reply:
We can be sure there will be the exact number of storage values as we requested. Neutron [takes care](https://github.com/neutron-org/neutron/blob/bf8b4f3e5b8075a0e461f04467e45af0525026ff/x/interchainqueries/keeper/msg_server.go#L99) of it.